guides/v2.2/coding-standards/technical-guidelines.md

```diff
@@ -532,7 +532,75 @@ class View extends Template
 
 ### 6.4. Service Contracts (Application) layer
 
-We are reviewing this section and will publish it soon.
+6.4.1. Location
+
+6.4.1.1. Service contract interfaces should be placed in separate API modules. All other modules will depend on the API module, while implementations could be easily swapped via `di.xml`. For example, if module is named `MyModule`, its APIs should be declared in the module named `MyModuleApi`.
+
+6.4.1.2. Service interfaces which should be exposed as web APIs must be placed under `MyModuleApi/Api/Service` directory. Service data interfaces must be placed under `MyModuleApi/Api/Data`. 
+
+6.4.1.3. All other APIs, including explicit extension points like Chain or Composite implementations must be placed under `MyModuleApi/Model`. 
+
+6.4.1.4. Arbitrary directory nesting can be used for APIs to improve code structure. For example, `MyCompany\MyModuleApi\Api\Service\Relations\DeleteInterface`.
```

Comment:
Suggest to call it RemoverInterface, name should be noun.

Reply:
@melnikovi 
>name should be a noun

Actually not. This rule is applicable for Classes which represent "things/entities", therefore you want to name them with nouns, and methods inside these classes represent actions over this entity so that methods usually represented by verbs. 

But this approach does not work with Functors (Function objects) - https://en.wikipedia.org/wiki/Function_object
And our new services are very close to Functors, because:
- they only consist of a single method
- the method name does not provide own semantic, it's always "execute"

In the beginning, we even wanted to use `__invoke` instead of `execute` to prevent redundancy.
https://twitter.com/iminyaylo/status/938122403443552256
But people voted to use `execute`.

So, if we deal with Functors it's much more natural to them to be represented by verbs, as functors represent  "action" by their nature.

You can read more about naming of factors here - https://stackoverflow.com/questions/10811169/do-you-use-nouns-for-classnames-that-represent-callable-objects